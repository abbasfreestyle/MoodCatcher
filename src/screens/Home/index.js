import React from 'react';
import PropTypes from 'prop-types';

import { View, StyleSheet } from 'react-native';

import { Button, Title, Background } from 'components';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'space-between'
  }
});

class Home extends React.Component {
  static navigationOptions = {
    title: 'Mood Catcher'
  };

  render() {
    const { navigation, screenProps } = this.props;
    console.log('screenProps', screenProps);
    const { username } = screenProps.authData;
    return (
      <Background>
        <View style={styles.container}>
          <Title>Hello {username}</Title>
          <Button.Card
            onPress={() => {
              navigation.navigate('SelectMood');
            }}
          >
            How are how you feeling?
          </Button.Card>
          <Button.Card
            onPress={() => {
              navigation.navigate('Dashboard');
            }}
          >
            Look back on my feelings
          </Button.Card>
        </View>
      </Background>
    );
  }
}

Home.propTypes = {
  navigation: PropTypes.object.isRequired,
  screenProps: PropTypes.object.isRequired
};

export default Home;
