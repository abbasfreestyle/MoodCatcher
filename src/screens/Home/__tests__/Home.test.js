import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Home from '..';

const requiredProps = {
  navigation: { navigate: jest.fn() },
  screenProps: { authData: { username: 'Test user' } }
};

const wrapper = shallow(<Home {...requiredProps} />);

describe('Screens > Home', () => {
  describe('Snapshots', () => {
    test('should match', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
