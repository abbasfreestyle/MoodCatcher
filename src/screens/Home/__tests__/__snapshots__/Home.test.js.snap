// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screens > Home Snapshots should match 1`] = `
<Background>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "space-between",
      }
    }
  >
    <Title>
      Hello Test user
    </Title>
    <Card
      onPress={[Function]}
    >
      How are how you feeling?
    </Card>
    <Card
      onPress={[Function]}
    >
      Look back on my feelings
    </Card>
  </View>
</Background>
`;
